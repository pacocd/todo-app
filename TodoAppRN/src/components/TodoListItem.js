import React, { Component } from 'react';
import { Text, View, Switch } from 'react-native';
import PropTypes from 'prop-types';

export class TodoListItem extends Component {
  static propTypes = {
    todo: PropTypes.shape({
      id: PropTypes.number.isRequired,
      user_id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      completed: PropTypes.bool.isRequired
    }).isRequired
  };

  render() {
    const { title } = this.props.todo;
    return (
      <View style={{ width: '100%', paddingVertical: 14 }}>
        <Text>{title}</Text>
      </View>
    );
  }
}

export default TodoListItem;
