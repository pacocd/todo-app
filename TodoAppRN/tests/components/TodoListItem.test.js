import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TodoListItem from '../../src/components/TodoListItem';

configure({ adapter: new Adapter() });

describe('when TodoListItem is fully renderd', () => {
  const props = {
    todo: { id: 1, user_id: 1, title: 'test', completed: false }
  };
  const wrapper = shallow(<TodoListItem {...props} />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
