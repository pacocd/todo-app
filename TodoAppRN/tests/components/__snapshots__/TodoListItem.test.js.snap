// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when TodoListItem is fully renderd renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoListItem
    todo={
      Object {
        "completed": false,
        "id": 1,
        "title": "test",
        "user_id": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        test
      </Text>,
      "style": Object {
        "paddingVertical": 14,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "test",
        "ellipsizeMode": "tail",
      },
      "ref": null,
      "rendered": "test",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          test
        </Text>,
        "style": Object {
          "paddingVertical": 14,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "test",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "test",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
